### ISSUE TITLE
Support for clustered vert.x in native image

### ISSUE THREAD

**Issue by prasannak-ra** [quarkusio/quarkus#35798]
### Description

PR https://github.com/quarkusio/quarkus/pull/3979 has disabled clustered vert.x in native image with the comment - "Disable clustered vert.x in native image as it does not work anyway."
We're using quarkus-based services which use clustered event bus and are exploring creating native binaries for the services.
But based on above PR, it looks like we cannot create native binaries till the time clustered vert.x is supported.
Please let me know, if there's any plan to support clustered vert.x in native image in near future?

**Comment by geoand** [comment]
@cescoffier I don't think we plan to do this, correct?

**Comment by cescoffier** [comment]
No, I didn't plan to have this yet. The most pressing issue is to have a cluster manager working reliably in a multi-node kube. Once we have this, that can be considered.

**Comment by prasannak-ra** [comment]
Thank Georgios and Clement for quick response. BTW, @cescoffier as far as our experience goes, we're running cluster manager (hazelcast based) in ~15 node AKS and have not really seen any issues. 
(I understand you may be referring to issues which I'm unaware of and community knows, just stating my observation.)

**Comment by prasannak-ra** [comment]
> No, I didn't plan to have this yet. The most pressing issue is to have a cluster manager working reliably in a multi-node kube. Once we have this, that can be considered.

Hi Clement, Can you please provide some info/links about instability of cluster manager in multi-node k8s?

**Comment by fbuetler** [comment]
Hi, we are also interested in deploying native image in clustered mode. Do you have any insights, why this `does not work anyway`, @cescoffier . Maybe the original committer @stuartwdouglas has some additional info?

**Comment by cescoffier** [comment]
@prasannak-ra when kubernetes starts moving pods around, the group communication and membership protocol gets lost. You hit many errors, and it takes time (minutes) to recover completely.

**Comment by cescoffier** [comment]
@fbuetler, there are multiple problems:

- that part of the Vert.x code would need to be adapted to work in native mode correctly
- the cluster manager implementation would need to work in native (and first, compile to native correctly and then behave correctly).

**Comment by prasannak-ra** [comment]
> @prasannak-ra when kubernetes starts moving pods around, the group communication and membership protocol gets lost. You hit many errors, and it takes time (minutes) to recover completely.

Thanks @cescoffier for the details. Now that you mentioned it, I remember this is something we witnessed in chaos testing. Then I was able to replicate the behavior by randomly killing/launching containers on my local.
We implemented following solution to stabilize the cluster (vert.x cluster) - 
1. Created an external HZ two-pod cluster running independently in k8s.  (with configuration, one backup for each partition)
2.  All the Quarkus based services which use clustered event bus, join as "[lite members]([https://docs.hazelcast.com/hazelcast/5.3/maintain-cluster/lite-members#:~:text=Lite%20members%20are%20the%20Hazelcast,members%20to%20run%20heavy%20computations.])" to HZ cluster.  
3. None of my services share any membership/registration data now, so when my service pods move around, there's no outage.

But this is only possible because of HZ lite member functionality, this won't work with inifinispan or ignite.
